File: dev/RecommenderAttacksAnalytics/src/main/java/com/codetrix/entities/common/AbstractItem.java

package com.codetrix.entities.common;

public abstract class AbstractItem implements IPersistenceEntity{
    
    protected long itemId;
    
    public long getId(){
        return this.itemId;
    }
    
    @Override
    public boolean equals(Object obj)
    {
        if( !(obj instanceof AbstractItem) )
            return false;
        
        if(obj == this)
            return true;
        
        return ((AbstractItem)obj).itemId == this.itemId;
    }
    
    @Override
    public int hashCode()
    {
        return Long.valueOf(this.itemId).hashCode();
    }
}

File: dev/CSharpVersion/RecommenderAttacksAnalytics/Entities/Common/AbstractItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace RecommenderAttacksAnalytics.Entities.Common
{
    public abstract class AbstractItem : AbstractPersistenceEntity
    {
        protected long m_itemId;

        public override long getId()
        {
            return m_itemId;
        }

    }
}
